src/transformers/models/informer/modeling_informer.py

```diff
@@ -0,0 +1,2429 @@
+# coding=utf-8
+# Copyright 2023 Amazon and The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch Informer model."""
+
+import random
+from dataclasses import dataclass
+from typing import Callable, Dict, List, Optional, Tuple, Union
+
+import numpy as np
+import torch
+from torch import nn
+from torch.distributions import (
+    AffineTransform,
+    Distribution,
+    Independent,
+    NegativeBinomial,
+    Normal,
+    StudentT,
+    TransformedDistribution,
+)
+
+from ...activations import ACT2FN
+from ...modeling_outputs import BaseModelOutput, BaseModelOutputWithPastAndCrossAttentions, ModelOutput
+from ...modeling_utils import PreTrainedModel
+from ...utils import add_start_docstrings, add_start_docstrings_to_model_forward, logging, replace_return_docstrings
+from .configuration_informer import InformerConfig
+
+
+logger = logging.get_logger(__name__)
+
+_CONFIG_FOR_DOC = "InformerConfig"
+
+
+INFORMER_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "huggingface/informer-tourism-monthly",
+    # See all Informer models at https://huggingface.co/models?filter=informer
+]
+
+
+# Copied from transformers.models.time_series_transformer.modeling_time_series_transformer.AffineTransformed
+class AffineTransformed(TransformedDistribution):
```

Comment:
All those copied from create two objects with the same names in different modeling files. This is why each of them should be prefixed with the name of the model (which also appies to time_series_transformer.

Reply:
@kashif I think it's better you will handle this